RFCs/0017-TestCase-FullyQualifiedName.md

```diff
@@ -1,49 +1,97 @@
+
+`{type}` - The `type` segment includes the fully specified type name in metadata format:
+
+    NamespaceA.NamespaceB.ClassName`1+InnerClass`2
+
+* The type name must be fully qualified (in the CLR sense), including its namespace. Any generic classes must also include an arity value using backtick notation (`# where # is the number of type arguments that the class requires).
+* Nested classes are appended with a '+' and must also include an arity if generic.
+* There should be no extra whitespace included in the segment
+
+`{method}` - The `method` segment includes the fully specified method including the method name and a list of its parameter types inside parentheses separated by commas.
+
+    MethodName`2(ParamTypeA,ParamTypeB,…)
+
+* If the method accepts no parameters, then the parentheses should be omitted.
+* If the method is generic, then an arity must be specified (in the same way as the type segment).
+* Insignificant whitespace should not be included in the segment.
+* The list of parameter types must be encoded in the same way that a type in the `{type}` segment of the FQN is encoded.
+
+## Custom Test Naming
+
+Some of the adapters currently modify the FQN when tests have custom names. This standard would disallow that. FQN can only represent Class/Method symbols, not test names. Test names should be reflected only in the DisplayName.
+
+## Syntactic vs Semantic Test Location
+
+We are defining syntactic location as the location in the code/syntax where the method is defined that implements the test. On the other hand, semantic location is the location in the type hierarchy at which point the method becomes recognized as a valid test. 
+
+There can be a situation where a test is declared on an abstract base type (syntactic), but the test is only discovered in a derived class (semantic). The question is which type should the FQN point to? The base type, where the test code is located? Or the derived type, where the test is discovered. We will use the semantic location for the type because that is where the test adapter will find the test and it is where the test class would be instantiated. Also, there is a one to many relationship between the two. There can be many semantic test locations that each are implemented by the same base class method. If we were to standardize on using the syntactic location, we would risk having name collisions or fail to find the correct test case since there are multiple possible candidates.
+
+## Managed Tests that are Not Type/Method Based
+
+It is possible to create unit tests in managed code where tests are not necessarily based on types & methods. While this is somewhat uncommon, we need to be able to handle this situation gracefully. Some examples are gherkin-dotnet and custom discoverers in xUnit. In these examples, an externally parsed definition file may determine the tests. For these cases, we have defined the 'other' form of FQN. FQNs in this form will look similar to the following.
```

Comment:
gherkin-dotnet is only the parser for Gherkin, which is used by SpecFlow. At SpecFlow we are generating coded tests in a code behind file (for xUnit, NUnit, MSTest or SpecFlow+Runner). The SpecFlow+Runner has a custom test discovery where not everything is only from the code, but also from configuration.

Reply:
@peterwald sorry for the late respone. The notification went under.
I think I only wanted to point out, that gherkin-dotnet is not "correct" here, as it is only the parser.

About the method-based FQNs & SpecFlow+Runner: I worked some weeks ago on an implementation of it and it looked good that we also can provide them. And as said, we always can fallback to the "other" method.